src/nrniv/kschan.cpp

```diff
@@ -2396,25 +2450,26 @@ void KSChan::state_consist(int shift) {  // shift when Nsingle winks in and out
             Prop* p;
             for (p = nd->prop; p; p = p->next) {
                 if (p->_type == mtype) {
-                    if (p->param_size != ns) {
+                    if (p->param_size() != ns) {
                         v_structure_change = 1;
-                        double* oldp = p->param;
-                        p->param = (double*) erealloc(oldp, ns * sizeof(double));
-                        if (oldp != p->param || shift != 0) {
-                            // printf("KSChan::state_consist realloc changed location\n");
-                            notify_freed_val_array(oldp, p->param_size);
-                        }
-                        p->param_size = ns;
-                        if (shift == 1) {
-                            for (j = ns - 1; j > 0; --j) {
-                                p->param[j] = p->param[j - 1];
-                            }
-                            p->param[0] = 1;
-                        } else if (shift == -1) {
-                            for (j = 1; j < ns; ++j) {
-                                p->param[j - 1] = p->param[j];
-                            }
-                        }
+                        assert(false);
```

Comment:
Evidently this line is not reached by any tests I have run.

Reply:
cc: @nrnhines

Reply:
#2080 supplies a test that covers this function